docs/website/docs/reference/frequently-asked-questions.md

```diff
@@ -0,0 +1,114 @@
+
+These methods allow for a degree of customization in handling data structure and loading processes, serving as interim solutions until a direct feature is developed.
+
+## Can I configure dlt to load data in chunks of 10,000 records for more efficient processing, and how does this affect data resumption and retries in case of failures?
+
+`dlt` buffers to disk, and has built-in resume and retry mechanisms. This makes it less beneficial to manually manage atomicity after the fact unless you're running serverless. If you choose to load every 10k records instead, you could potentially see benefits like quicker data arrival if you're actively reading, and easier resumption from the last loaded point in case of failure, assuming that state is well-managed and records are sorted.
+
+It's worth noting that `dlt` includes a request library replacement with built-in retries. This means if you pull 10 million records individually, your data should remain safe even in the face of network issues. To resume jobs after a failure, however, it's necessary to run the pipeline in its own virtual machine (VM). Ephemeral storage solutions like Cloud Run don't support job resumption.
+
+## How to contribute a verified source?
+
+To submit your source code to our verified source repository, please refer to [this guide](https://github.com/dlt-hub/verified-sources/blob/master/CONTRIBUTING.md).
+
+## If you don't have the source required listed in verified sources?
+
+In case you don't have the source required listed in the [verified sources](../../docs/dlt-ecosystem/verified-sources/), you could create your own pipeline by refering to the [following docs](../../docs/walkthroughs/create-a-pipeline). 
+
+## How can I retrieve the complete schema of a `dlt` source?
+
+To retrieve the complete schema of a `dlt` source as `dlt` recognizes it, you need to execute both the extract and normalization steps. This process can be conducted on a small sample size to avoid processing large volumes of data. You can limit the sample size by applying an `add_limit` to your resources. Here is a sample code snippet demonstrating this process:
+
+```py
+# Initialize a new DLT pipeline with specified properties
+p = dlt.pipeline(
+    pipeline_name="my_pipeline",
+    destination="duckdb",
+    dataset_name="my_dataset"
+)
+
+# Extract data using the predefined source `my_source`
+p.extract(my_source().add_limit(10))
+
+# Normalize the data structure for consistency
+p.normalize()
+
+# Print the default schema of the pipeline in pretty YAML format for review
+print(p.default_schema.to_pretty_yaml())
+```
+
+This method ensures you obtain the full schema details, including all columns, as seen by `dlt`, without needing a predefined contract on these resources.
+
+## Can `dlt` periodically commit updates to a resource state to an external database during a pipeline run?
```

Comment:
I think a different heading is need. It would make reading the heading better right now it doesn't make sense to me. @AstrakhantsevaAA 

>Can `dlt` periodically commit updates to a resource state, and upload/ ingest to an external database during a pipeline run?

Reply:
yes, changed.

Reply:
this question is about extract step? or it's particular about remote destination? or it;s about state in common?

Reply:
This question is about committing the info for the extract step to the state so that if the pipeline fails, the user can know where it failed. I improved the FAQ and explanation.